### DISCUSSION TITLE
Flash only firmware on ESP32S3 via USB2 port

### DISCUSSION THREAD

**Discussion by mforbiz** [openthread/openthread#9993] [Q&A] [answered]
Hi.
Following this:
https://openthread.io/guides/border-router/espressif-esp32#build-and-run-the-border-router
it seems to me that I only must flash firmware on ESP32S3 via USB2 port and not on the ESP32-H2 chip (via USB1) port because it will be flashed at the next reboot.
Is this right?
Thanks in advance

**Comment by jwhui** [comment]
As mentioned in the guide you linked:
> The firmware doesn't need to be explicitly flashed to a device. It will be included in the Border Router firmware and flashed to the ESP32-H2 chip upon first boot (or the RCP firmware changed).

**Reply by mforbiz** [reply]
Hi jwhui.
Thanks for you reply.
When I flash ESP32S3 by means:
`idf.py -p  flash monitor`
must I specify the USB2 port (_anything like /dev/ttyACM1_)
so that the right command will be:
`idf.py -p  /dev/ttyACM1 flash monitor` ?
And second question: must I use simultaneously two USB C cable connected to the board USB1 ans USB2 ?
![2024-04-04_165834](https://github.com/openthread/openthread/assets/105542757/47bb2d9e-c1a8-4201-87f8-bde4e0d641b0)

**Reply by jwhui** [reply]
> must I specify the USB2 port (anything like /dev/ttyACM1)

As stated in: https://openthread.io/codelabs/esp-openthread-hardware#2:

```
$ idf.py -p <your-local-port> build flash monitor
```

> And second question: must I use simultaneously two USB C cable connected to the board USB1 ans USB2 ?

I just replied to your same question here: https://github.com/orgs/openthread/discussions/9994#discussioncomment-9011551

**Reply by mforbiz** [reply]
Hi jwui.
I've configured all system as your suggestions but during the flash I've this error:
![error](https://github.com/openthread/openthread/assets/105542757/1184085a-42d6-47fb-b316-cbfe13a44b1b)
Any idea?

**Reply by chshu** [reply, marked as answer]
@mforbiz If maybe caused by some USB connection issue. Could you please try the following steps:

1. build the example first: `idf.py build`
2. try to put the device into the downloading mode manually: hold the `BOOT2` button, click the `RESET2` button, then release the `BOOT2` button
3. flash the firmware with this command:
```esptool.py -p /dev/ttyACM0 --no-stub -b 460800 --before default_reset --after hard_reset --chip esp32s3  write_flash --flash_mode dio --flash_size 4MB --flash_freq 80m 0x0 build/bootloader/bootloader.bin 0x8000 build/partition_table/partition-table.bin 0xf000 build/ota_data_initial.bin 0x20000 build/esp_ot_br.bin 0x340000 build/web_storage.bin 0x359000 build/rcp_fw.bin```

**Reply by mforbiz** [reply]
Hi chshu.
It seems to works fine!
Many thanks
